spec/integration/apply_spec.rb

```diff
@@ -67,13 +67,6 @@
     # Run tests that require the puppet-agent package to be installed on the target.
     # Each test is run against all agent targets unless otherwise noted.
     context 'with puppet installed' do
-      # Install agents on the agent targets.
-      before(:all) do
-        TEST_VERSIONS.each do |version, collection|
```

Comment:
I think this can also refactor tests that install and uninstall the agent on the main container to use the containers with agents instead, like I see that in `spec/integration/plan_spec.rb#L73` we install and uninstall the agent around each test instead of just using the containers with the agent installed. Or same in `spec/integration/device_spec.rb`. I think we still need the `uninstall` utility to uninstall after tests that verify installing the agent with apply_prep works, but we should be able to remove the `install` spec helper function with this PR.

Reply:
The `install` helper is still needed by the apply integration tests on Windows. Updated the plan and device integration tests to use the agentful containers.